docs/scos/dev/guidelines/keeping-a-project-upgradable/supported-extension-scenarios/modules-configuration.md

```diff
@@ -0,0 +1,256 @@
+- title: Event subscribers registration
+  link: docs/scos/dev/guidelines/keeping-a-project-upgradable/supported-extension-scenarios/event-subscribers-registration.html
+- title: Modules configuration
+  link: docs/scos/dev/guidelines/keeping-a-project-upgradable/supported-extension-scenarios/modules-configuration.html
+---
+
+## Introduction
+
+Manifests support changes in module configuration files.
+
+Manifests support all scalar types (bool, int, float, string), compound type array and special type null.
+
+Manifests do not support compound types object, callable, iterable and special type resource.
+
+Manifest only add the values to configuration files. Manifests DO NOT REMOVE the values from project configuration. 
+
+## 1.1. Basic scalar values as return
+
+Manifests fully support 4 scalar types:
+* bool
+* int
+* float (floating-point number)
+* string
+
+Manifests also support usage of the constants and have built in basic support for constant concatenation.
+
+Code example 1.1.1: Method returns string
+```php
+use Spryker\Client\RabbitMq\RabbitMqConfig as SprykerRabbitMqConfig;
+
+class RabbitMqConfig extends SprykerRabbitMqConfig
+{
+    ...
+    protected function getDefaultBoundQueueNamePrefix(): string
+    {
+        return 'error';
+    }
+}
+```
+
+Code example 1.1.2: Method returns string (constant used)
+```php
+use Pyz\Zed\Synchronization\SynchronizationConfig;
+use Spryker\Zed\ConfigurableBundleStorage\ConfigurableBundleStorageConfig as SprykerConfigurableBundleStorageConfig;
+
+class ConfigurableBundleStorageConfig extends SprykerConfigurableBundleStorageConfig
+{
+    ...
+    public function getConfigurableBundleTemplateSynchronizationPoolName(): string
+    {
+        return SynchronizationConfig::DEFAULT_SYNCHRONIZATION_POOL_NAME;
+    }
+}
+```
+
+Code example 1.1.3: Method returns string with constant concatenation
+```php
+use Spryker\Zed\Development\DevelopmentConfig as SprykerDevelopmentConfig;
+
+class DevelopmentConfig extends SprykerDevelopmentConfig
+{
+    ...
+    public function getCodingStandard(): string
+    {
+        return APPLICATION_ROOT_DIR . DIRECTORY_SEPARATOR . 'phpcs.xml';
+    }
+}
+```
+
+Code example 1.1.4: Method returns boolean
+```php
+use SprykerShop\Yves\ProductReplacementForWidget\ProductReplacementForWidgetConfig as SprykerShopProductReplacementForWidgetConfig;
+
+class ProductReplacementForWidgetConfig extends SprykerShopProductReplacementForWidgetConfig
+{
+    ...
+    public function isProductReplacementFilterActive(): bool
+    {
+        return true;
+    }
+}
+```
+
+Code example 3.1.5: Method returns int
+```php
+use Spryker\Zed\Customer\CustomerConfig as SprykerCustomerConfig;
+
+class CustomerConfig extends SprykerCustomerConfig
+{
+    ...
+    public function getCustomerPasswordSequenceLimit(): ?int
+    {
+        return 3;
+    }
+}
+```
+
+## 1.2. Array as return
+
+### 1.2.1. Indexed array as return
+
+Code example 3.2.1: Method returns indexed array
```

Comment:
please recheck the code example blocks numbering.

Reply:
Thanks fixed